components/ai_chat/content/browser/ai_chat_tab_helper.h

```diff
@@ -6,179 +6,39 @@
+#include "brave/components/ai_chat/content/browser/associated_web_contents.h"
 #include "content/public/browser/navigation_handle.h"
-#include "content/public/browser/render_frame_host.h"
 #include "content/public/browser/web_contents.h"
 #include "content/public/browser/web_contents_observer.h"
 #include "content/public/browser/web_contents_user_data.h"
-#include "pdf/buildflags.h"
-#include "url/gurl.h"
-
-#if BUILDFLAG(ENABLE_PDF)
-#include "pdf/mojom/pdf.mojom.h"
-#endif  // BUILDFLAG(ENABLE_PDF)
-
-namespace content {
-class NavigationEntry;
-class RenderFrameHost;
-class WebContents;
-struct LoadCommittedDetails;
-}  // namespace content
-
-class AIChatUIBrowserTest;
 
 namespace ai_chat {
-class AIChatMetrics;
 
 // Provides context to an AI Chat conversation in the form of the Tab's content
-class AIChatTabHelper : public content::WebContentsObserver,
-                        public content::WebContentsUserData<AIChatTabHelper>,
-                        public AssociatedContentDriver {
+class AIChatTabHelper : public content::WebContentsUserData<AIChatTabHelper> {
  public:
-  // Delegate to extract print preview content
-  class PrintPreviewExtractionDelegate {
-   public:
-    // Result is image data of pages or error
-    using CaptureImagesCallback = base::OnceCallback<void(
-        base::expected<std::vector<std::vector<uint8_t>>, std::string>)>;
-
-    virtual ~PrintPreviewExtractionDelegate() = default;
-    // Capture images of content without doing OCR
-    virtual void CaptureImages(CaptureImagesCallback callback) = 0;
-  };
-
-  class PageContentFetcherDelegate {
-   public:
-    virtual ~PageContentFetcherDelegate() = default;
-
-    // Gets text of the page content, making an attempt
-    // to only consider the main content of the page.
-    virtual void FetchPageContent(std::string_view invalidation_token,
-                                  FetchPageContentCallback callback) = 0;
-
-    // Attempts to find a search summarizer key for the page.
-    virtual void GetSearchSummarizerKey(
-        GetSearchSummarizerKeyCallback callback) = 0;
-
-    // Fetches the nonce for the OpenLeo button from the page HTML and validate
-    // if it matches the href URL and the passed in nonce.
-    virtual void GetOpenAIChatButtonNonce(
-        mojom::PageContentExtractor::GetOpenAIChatButtonNonceCallback
-            callback) = 0;
-  };
-
   AIChatTabHelper(const AIChatTabHelper&) = delete;
   AIChatTabHelper& operator=(const AIChatTabHelper&) = delete;
   ~AIChatTabHelper() override;
 
-  void SetPageContentFetcherDelegateForTesting(
-      std::unique_ptr<PageContentFetcherDelegate> delegate) {
-    page_content_fetcher_delegate_ = std::move(delegate);
-  }
-  raw_ptr<PageContentFetcherDelegate>
-  GetPageContentFetcherDelegateForTesting() {
-    return page_content_fetcher_delegate_.get();
+  AssociatedWebContents* associated_web_contents() const {
+    return associated_web_contents_.get();
```

Comment:
If I understand correctly, we currently don't expect this to return nullptr, should we make this returning a reference instead?